src/module/actor/sheet/spellcasting-dialog.ts

```diff
@@ -47,36 +47,51 @@ class SpellcastingCreateAndEditDialog extends FormApplication<Embedded<Spellcast
         return {
             ...(await super.getData()),
             actor: this.actor,
-            data: this.object.toObject(false).system,
+            data: this.object.toObject(true).system,
             magicTraditions: CONFIG.PF2E.magicTraditions,
             spellcastingTypes: CONFIG.PF2E.preparationType,
             abilities: CONFIG.PF2E.abilities,
+            hasAbility: this.#canSetAbility(),
         };
     }
 
+    /** Returns whether or not the spellcasting data can include an ability */
+    #canSetAbility(): boolean {
+        const slug = this.object._source.system.proficiency.slug;
+        const baseStat = this.actor.isOfType("character") ? this.actor.getProficiencyStatistic(slug) : null;
+        return !slug || (!!baseStat && !baseStat.ability);
+    }
+
     protected override async _updateObject(event: Event, formData: Record<string, unknown>): Promise<void> {
         const wasInnate = this.object.isInnate;
 
         // Unflatten the form data, so that we may make some modifications
         const inputData: DeepPartial<SpellcastingEntrySource> = expandObject(formData);
-        const system = inputData.system;
 
         // We may disable certain form data, so reinject it
-        inputData.system = mergeObject(
+        const system = mergeObject(
             inputData.system ?? {},
             {
                 prepared: {
                     value: this.object.system.prepared.value,
                 },
+                ability: { value: "cha" },
```

Comment:
Ability is cha?

Reply:
Charisma is the default ability as defined by template.json. This is needed since ability can disappear.

It'd be better to make a best guess at what ability to use, but that's a separate feature and a separate PR.